Tasks/Maven/maven2.js

```diff
@@ -169,22 +169,21 @@ will still succeed but the report will have less data.
 */
 
 
-var runFailed = false;
+var userRunFailed = false;
+var sqRunFailed = false;
 
 mvnv.exec()
 .fail(function (err) {
     console.error("Maven is not installed on the agent");
-    tl.exit(1);
+    tl.exit(1);  // tl.exit sets the step result but does not stop execution
+    process.exit(1);
 })
 .then(function (code) {
     return mvnb.exec(); // run Maven with the user specified goals
 })
 .fail(function (err) {
     console.error(err.message);
-    runFailed = true; // record the error, but do not exit
-})
-.fin(function () {
-    publishTestResults(publishJUnitResults, testResultsFiles); // publish test results even if tests fail, causing Maven to fail
+    userRunFailed = true; // record the error and continue
 })
 .then(function (code) {
     mvnsq = getSonarQubeRunner();
```

Comment:
If SQ is not configured, mvnsq will be null and this promise will return. This is supported by Q promises and it doesn't break the promise chain - see https://strongloop.com/strongblog/promises-in-node-js-with-q-an-alternative-to-callbacks/ (Promises and synchronous functions). The subsequent .fail and .then clauses will be hit.